### PR TITLE
Improve credential how-to description in README 

### PR DESCRIPTION
**Pull request by PeterFogh**
I have used adlfs the last year using `account_key` but found out today that Azure automatically can find available credentials when using `storage_options={'account_name': ACCOUNT_NAME, 'anon': False}`. I thought this could be documented more clearly, thus I like to add these changes to the docs.

Thanks to @charles-mccall for helping me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,32 +68,33 @@ with suitable credentials to perform operations on the resources of choice.
...
+The following environmental variables can also be set and picked up for authentication:
+- "AZURE_STORAGE_CONNECTION_STRING"
+- "AZURE_STORAGE_ACCOUNT_NAME"
+- "AZURE_STORAGE_ACCOUNT_KEY"
+- "AZURE_STORAGE_SAS_TOKEN"
+- "AZURE_STORAGE_TENANT_ID"
+- "AZURE_STORAGE_CLIENT_ID"
+- "AZURE_STORAGE_CLIENT_SECRET"
+
+The filesystem can be instantiated for different use cases base on a variety of `storage_options` combinations. The following list describes some commen use cases utilizing `AzureBlobFileSystem`, i.e. protocols `abfs`or `az`. Note all cases require the `account_name` argument to be provided:
```
Comment (efiop) [inline README.md:92]:
```suggestion
The filesystem can be instantiated for different use cases based on a variety of `storage_options` combinations. The following list describes some common use cases utilizing `AzureBlobFileSystem`, i.e. protocols `abfs`or `az`. Note that all cases require the `account_name` argument to be provided:
```

File: README.md [hunk 2]
```diff
@@ -19,7 +19,6 @@ or
 The `adl://` and `abfs://` protocols are included in fsspec's known_implementations registry 
 in fsspec > 0.6.1, otherwise users must explicitly inform fsspec about the supported adlfs protocols.
 
-
```
Comment (efiop) [inline README.md:22]: Unrelated change.

File: README.md [hunk 3]
```diff
@@ -70,32 +68,33 @@ with suitable credentials to perform operations on the resources of choice.
...
+1. Anonymous connection to public container: `storage_options={'account_name': ACCOUNT_NAME, 'anon': True}` will assume the `ACCOUNT_NAME` points to a public container, and attempt to use an anonymous login. Note, the default value for `anon` is True.
+2. Auto credential solving using Azure's DefaultAzureCredential() library: `storage_options={'account_name': ACCOUNT_NAME, 'anon': False}` will use [`DefaultAzureCredential`](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) to get valid credentials to the container `ACCOUNT_NAME`. `DefaultAzureCredential` attempts to authenticate via the [mechanisms and order visualized here](https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python#defaultazurecredential).
+3. Azure ServicePrincipal: `tenant_id`, `client_id`, and `client_secret` are all used as crendentials for a Azure ServicePrincipal: e.g. `storage_options={'account_name': ACCOUNT_NAME, 'tenant_id': TENANT_ID, 'client_id': CLIENT_ID, 'client_secret': CLIENT_SECRET}`.
```
Comment (efiop) [inline README.md:95]:
```suggestion
3. Azure ServicePrincipal: `tenant_id`, `client_id`, and `client_secret` are all used as credentials for an Azure ServicePrincipal: e.g. `storage_options={'account_name': ACCOUNT_NAME, 'tenant_id': TENANT_ID, 'client_id': CLIENT_ID, 'client_secret': CLIENT_SECRET}`.
```

**Review by efiop — CHANGES_REQUESTED** [review]
Some minor typos, but looks great to me overall.

@PeterFogh Long time no see ;) Hope you are doing great.

**Comment by hayesgb** [comment]
LGTM!  Thanks @PeterFogh !

**Pull request merged**